extensions/mqtt/processors/AbstractMQTTProcessor.h

```diff
@@ -44,47 +59,76 @@ class AbstractMQTTProcessor : public core::Processor {
     freeResources();
   }
 
-  SMART_ENUM(MqttVersions,
-    (V_3X_AUTO, "3.x AUTO"),
-    (V_3_1_0, "3.1.0"),
-    (V_3_1_1, "3.1.1"),
-    (V_5_0, "5.0"));
-
-  SMART_ENUM(MqttQoS,
-    (LEVEL_0, "0"),
-    (LEVEL_1, "1"),
-    (LEVEL_2, "2"));
-
-  EXTENSIONAPI static const core::Property BrokerURI;
-  EXTENSIONAPI static const core::Property ClientID;
-  EXTENSIONAPI static const core::Property QoS;
-  EXTENSIONAPI static const core::Property MqttVersion;
-  EXTENSIONAPI static const core::Property ConnectionTimeout;
-  EXTENSIONAPI static const core::Property KeepAliveInterval;
-  EXTENSIONAPI static const core::Property LastWillTopic;
-  EXTENSIONAPI static const core::Property LastWillMessage;
-  EXTENSIONAPI static const core::Property LastWillQoS;
-  EXTENSIONAPI static const core::Property LastWillRetain;
-  EXTENSIONAPI static const core::Property LastWillContentType;
-  EXTENSIONAPI static const core::Property Username;
-  EXTENSIONAPI static const core::Property Password;
-  EXTENSIONAPI static const core::Property SecurityProtocol;
-  EXTENSIONAPI static const core::Property SecurityCA;
-  EXTENSIONAPI static const core::Property SecurityCert;
-  EXTENSIONAPI static const core::Property SecurityPrivateKey;
-  EXTENSIONAPI static const core::Property SecurityPrivateKeyPassword;
-
-
-  static auto basicProperties() {
-    return std::array{
+  EXTENSIONAPI static constexpr auto BrokerURI = core::PropertyDefinitionBuilder<>::createProperty("Broker URI")
+      .withDescription("The URI to use to connect to the MQTT broker")
+      .isRequired(true)
+      .build();
+  EXTENSIONAPI static constexpr auto ClientID =       core::PropertyDefinitionBuilder<>::createProperty("Client ID")
```

Comment:
Extra spaces

Reply:
fixed in bde74b220d3a6755b91e5104648b0a2b0f57fafc